include/core/mir/mutex.h

```diff
@@ -33,11 +33,18 @@ class MutexGuard
 {
 public:
     MutexGuard(std::unique_lock<std::mutex>&& lock, Guarded& value)
```

Comment:
Hm. So, the names were stolen from the Rust standard library; this is how `Mutex<T>` works there.

I'm not averse to renaming them, but I don't think `Secure<State>` is a better name. Particularly, it doesn't describe what it's securing the state *from*, and there are various prominent `SecureFoo` things (`SecureString` being I think the most notable) which are information-security related which this might get confused with.

This class ensures that all accesses to the contained data are coherent, in that you never see partially-updated state. Perhaps `Synchronised<State>` and `Synchronised<State>::Accessor` (moving `MutexGuard` into an inner type) for the two types?

It's also kinda important that the *implementation* is mutual-exclusion rather than, for example, atomic CAS? Is `Mutex<T>` is going to be confusing to anyone?

Reply:
Unfortunately, prior art is no guarantee of good names. None of the suggestions so far (including mine) are clearly good. Maybe `Synchronised` and `SyncLock`?

Anyway, in the absence of a clearly better alternative I won't block on these names.

Reply:
`Synchronized<State>::lock() -> Unique<State>`? `Locked` and `SyncLock` also fine. I'm not hugely particular about the names.

Reply:
I think I'd go for `Synchronized<State>::lock() -> Locked<State>`.

Reply:
Good news! It turns out we've actually *already* had this naming discussion, and `mir::Synchronised<T>` already exists! (servicing `LinuxVTProvider`).

I think we've approximate consensus on `Synchonised`/`Synchronised::Locked`, so I've updated the naming to that and folded in the exsiting, near identical  `src/include/server/mir/synchronised.h`.

Reply:
Wow! The previous naming discussion #384 was remarkably similar. I guess none of us learnt from the experience.